tests/queries/test_qs_combinators.py

```diff
@@ -234,6 +234,13 @@ def test_union_with_values_list_and_order_on_annotation(self):
             operator.itemgetter('num'),
         )
 
+    def test_union_multi_model_with_order(self):
+        ReservedName.objects.create(name='rn', order=0)
+        qs1 = Celebrity.objects.all()
+        qs2 = ReservedName.objects.all()
+        qs = qs1.union(qs2).values_list('name', flat=True)
+        self.assertEqual(qs.first(), 'rn')
```

Comment:
`first()` is not crucial for this regression, so I've changed this to the `order_by()`.

Reply:
Yes, the issue ia with order by.